scenario.py

```diff
@@ -423,14 +436,32 @@ def split_data(self, is_logging_enabled=True):
                 )
             splitting_indices_train = (splitting_indices * len(y_train)).astype(int)
 
+        # If partners don't use the full dataset we only keep the amount of data targeted and then normalize proportions
+
+        proportion_of_dataset = np.sum(self.amounts_per_partner)
+
+        if proportion_of_dataset < 1:
+            skip_idx = int(round(len(x_train) * proportion_of_dataset))
+            train_idx = np.arange(len(x_train))
+            np.random.seed(42)
+            np.random.shuffle(train_idx)
+            x_train = x_train[train_idx[0:skip_idx]]
+            y_train = y_train[train_idx[0:skip_idx]]
+
+            splitting_indices = splitting_indices / proportion_of_dataset  # Normalize proportions
+            if self.partners_count == 1:
+                splitting_indices_train = 1  # Read Documentation of np.split
```

Comment:
Could you elaborate :) ?

Reply:
I removed this line because it's useless,
I wrote the comment because I was surprised `np.split` could take as a argument, an array or an integer(1).
`np.split(train_idx, 1)` return the array train_idx

Reply:
Could you write this explanation directly in the comment?